docs/changes/coordinates/13484.bugfix.rst

```diff
@@ -0,0 +1,5 @@
+Deprecate the ``BaseFrame.get_frame_attr_names`` method,
+which actually returns an expensive to build up dict also providing
+the default values of the attributes.
+If you only need the names, use ``BaseFrame.frame_attributes.keys()``,
+if you need the defaults, use the new ``BaseFrame.get_frame_attr_defaults`` instead.
```

Comment:
I think the current entry is too focused on implementation details. End users might not be too interested about some `dict` being slow to build.
```suggestion
``BaseCoordinateFrame.get_frame_attr_names()`` had a misleading name,
because it actually provided a ``dict`` of attribute names and
their default values. It is now deprecated and replaced by ``BaseCoordinateFrame.get_frame_attr_defaults()``.
The fastest way to obtain the attribute names is ``BaseFrame.frame_attributes.keys()``.
```